File: 205-isomorphic-strings/isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        hash={}
        
        for sc,tc in zip(s,t):
            if sc in hash:
                if hash[sc]!= tc:
                    return False
            elif tc in hash.values():
                return False
            
            hash[sc]=tc
        return True
        

File: 205-isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char,int> mp;
        unordered_map<char,int> mp1;
        for(int i=0;i<s.size();i++){
        if(mp.find(s[i])==mp.end())
        mp[s[i]]=i;

        if(mp1.find(t[i])==mp1.end())
        mp1[t[i]]=i;

        if(mp[s[i]]!=mp1[t[i]])
        return false;
        }

return true;
        
    }
    
        
    
};